src/convertToObject.js

```diff
@@ -3,16 +3,41 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example in
+ * [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splitted = sourceString.split(';');
+  const splittedKeysAndValues = [];
+  const keysAndValues = {};
+
+  for (const item of splitted) {
+    splittedKeysAndValues.push(item.split(':'));
+  }
+
+  const splittedTrim = splittedKeysAndValues
+    .filter((arr) => arr.length > 1)
+    .map((keyAndValue) =>
+      [
+        keyAndValue[0].trim(),
+        keyAndValue[1].trim(),
+      ]
+    );
+
+  for (const arr of splittedTrim) {
```

Comment:
Instead of creating a new array 'splittedTrim' and then iterating over it again, you could simply add each key-value pair to 'keysAndValues' inside the first loop. This would make the code more efficient.